### PR TITLE
Add ZipResultStep

### PR DESCRIPTION
**Pull request by kretep**
to make sure the zip file is generated, even after the user sends the task to the background and logs out.

### CODE REVIEW

**Comment by kretep** [comment]
@forus could you have a look at this?

File: src/groovy/jobs/steps/ZipResultsStep.groovy [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Override
+    void execute() {
+
+        String analysisDirectory = tempFolderDirectory + jobName + File.separator
+        this.tempDirectory = analysisDirectory + "workingDirectory" + File.separator
+
+        try {
+            boolean isAllowedToExport = asyncJobService.isUserAllowedToExportResults(currentUserBean, jobName)
```
Comment (forus) [inline src/groovy/jobs/steps/ZipResultsStep.groovy]: This step could be skipped altogether if user is not allowed to export results.

File: src/groovy/jobs/steps/ZipResultsStep.groovy [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            if (isAllowedToExport) {
+                // Zip the working directory
+                String zipLocation = "${analysisDirectory}zippedData.zip"
+                if (!new File(zipLocation).isFile()) {
```
Comment (forus) [inline src/groovy/jobs/steps/ZipResultsStep.groovy]: I think we do not need this check anymore.

File: src/groovy/jobs/steps/ZipResultsStep.groovy [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                    zipService.zipFolder(tempDirectory, zipLocation)
+                }
+            }
+        } catch (Exception e) {
```
Comment (forus) [inline src/groovy/jobs/steps/ZipResultsStep.groovy]: Empty try-catch block is never a good idea. I think we should not catch exception here anyway. It's something we could do where we iterate over steps list.

**Comment by forus** [comment]
@kretep Please write unit tests for `ZipResultsStep`.

**Comment by forus** [comment]
@kretep There is another PR that has to fix the issue. https://github.com/CTMM-TraIT/Rmodules/pull/3

**Comment by kretep** [comment]
@forus Ok, but why does it seem that #2 was merged instead of #3?

**Pull request merged**